ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -0,0 +1,376 @@
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* device_info_service =
+      DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+
+  if (!device_info_service)
+    return nullptr;
+
+  return device_info_service->GetLocalDeviceInfoProvider()->GetLocalDeviceInfo();
+}
+
+std::vector<std::unique_ptr<syncer::DeviceInfo>>
+BraveSyncWorker::GetDeviceList() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* device_info_service =
+      DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+
+  if (!device_info_service)
+    return std::vector<std::unique_ptr<syncer::DeviceInfo>>();
+
+  syncer::DeviceInfoTracker* tracker =
+      device_info_service->GetDeviceInfoTracker();
+  return tracker->GetAllDeviceInfo();
+}
+
+std::string BraveSyncWorker::GetOrCreateSyncCode() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* sync_service = GetSyncService();
+  std::string sync_code;
+  if (sync_service) {
+    sync_code = sync_service->GetOrCreateSyncCode();
+  }
+  return sync_code;
+}
+
+bool BraveSyncWorker::SetSyncCode(const std::string& sync_code) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  if (sync_code.empty())
+    return false;
+
+  auto* sync_service = GetSyncService();
+  if (!sync_service || !sync_service->SetSyncCode(sync_code)) {
+    return false;
+  }
+  return true;
+}
+
+std::string BraveSyncWorker::GetSyncCodeFromHexSeed(const std::string& hex_code_seed) {
+  DCHECK(!hex_code_seed.empty());
+
+  std::vector<uint8_t> bytes;
+		std::string sync_code_words;
+		if (base::HexStringToBytes(hex_code_seed, &bytes)) {
+				DCHECK_EQ(bytes.size(), SEED_BYTES_COUNT);
+				if (bytes.size(), SEED_BYTES_COUNT) {
+						sync_code_words = brave_sync::crypto::PassphraseFromBytes32(bytes);
+						if (sync_code_words.empty()) {
+								VLOG(1) << __func__ << " PassphraseFromBytes32 failed for " << hex_code_seed;
+						}
+				} else {
+						LOG(ERROR) << "wrong seed bytes " << bytes.size();
+				}
+				
+				DCHECK_NE(sync_code_words, "");
+		} else {
+				VLOG(1) << __func__ << " HexStringToBytes failed for " << hex_code_seed;
+		}
+		return sync_code_words;
+}
+
+bool BraveSyncWorker::IsFirstSetupComplete() {
+		syncer::SyncService* sync_service = GetSyncService();
+		return sync_service &&
+										sync_service->GetUserSettings()->IsFirstSetupComplete();
+}
+
+bool BraveSyncWorker::ResetSync() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* sync_service =
+      ProfileSyncServiceFactory::GetForBrowserState(browser_state_);
+
+  // Do not send self deleted commit if engine is not up and running
+  if (!sync_service || sync_service->GetTransportState() !=
+          syncer::SyncService::TransportState::ACTIVE) {
+    OnLocalDeviceInfoDeleted();
+    return true;
+  }
+
+  auto* local_device_info = GetLocalDeviceInfo();
+  if (!local_device_info) {
+    // May happens when we reset the chain immediately after connection
+    VLOG(1) << __func__ << " no local device info, cannot reset sync now";
```

Comment:
Do we have VLOG routing through to regular BraveRewards logging? Or do we have to setup something similar in Sync

Reply:
there is no need for anything like BraveRewards logging because this is now using the internal chromium logging mechanisms. We could change the ledger to use VLOG directly after this is merged cc @NejcZdovc @tmancey

Reply:
Yes but do internal chromium logging mechanisms log to console correctly for iOS?

Reply:
yes

Reply:
although I'm not 100% sure how to enable VLOG when running on ios